### ISSUE TITLE
Localization not working when using component inheritance

### ISSUE THREAD

**Issue by k-vekos** [robisim74/angular-l10n#165]
**I'm submitting a...**
```
[ ] bug report
[ ] feature request
[x] support request
```
I am having an issue when using Component inheritance where items are not being properly localized.

I am using [this technique](https://stackoverflow.com/questions/33970645/how-to-extend-a-component-with-dependency-injection-in-angular-2/40592524#40592524) to avoid having to re-inject `TranslationService` in the derived component when it should already be injected as part of the superclass.

Here is my StackBlitz example: https://stackblitz.com/edit/angular-l10n-ehdj29

**Expected behavior**
Under *derived.component:* it should say "Hello World!" just like it does under *home.component:*. 

**Actual behavior**
It remains untranslated and displays the resource key as if I had not configued a language/locale.

**Steps to reproduce the behavior**
Run the example app.

**Comment by robisim74** [comment]
Hi @k-vekos,

I honestly did not understand what you're trying to do, but surely the `ServiceLocator` does not do anything, because it does not create any instances:

```
export class ServiceLocator {
    static injector: Injector; // it is null...
}
```


This library already implements a `"ServiceLocator"`(https://github.com/robisim74/angular-l10n/blob/master/src/models/injector-ref.ts), which you can also use directly if you want:

```
...
import { InjectorRef } from 'angular-l10n';

@Component({})
export class BaseComponent {
  ...
  protected translationService: TranslationService;

  constructor() {
    this.translationService = InjectorRef.get(TranslationService);
    ...
  }
}
```

Then I suggest you do not use `@Language` and `@DefaultLocale` decorators in your `BaseComponent` (because you'll need to implement `OnInit` hook that could be overridden by the subclass): this library already provides the `Localization` class: https://github.com/robisim74/angular-l10n/blob/master/src/services/localization.ts that you can extend: https://robisim74.github.io/angular-l10n/spec/getting-the-translation/#pure-pipes > _Translation & Localization classes_

Greetings

**Comment by k-vekos** [comment]
I removed `ServiceLocator` and instead used the provided `InjectorRef` and now it is working as expected. Thank you for the help, I am trying to get angular-l10n to play nicely with this other Angular library called `devextreme-angular` and since their widgets are generated from code without any angular-style templates I'm unable to use pipes or the `translate` directive like I would in a regular Angular widget.

It would be great if I could come up with a solution to make it work with angular-l10n because otherwise they only support `ngx-translate` which doesn't provide as much functionality. Here's a related issue I opened on their github project just FYI: https://github.com/DevExpress/devextreme-angular/issues/606
